temporaldata/temporaldata.py

```diff
@@ -1684,6 +1684,30 @@ def __setattr__(self, name, value):
             # or not
             self._sorted = None
 
+    def __iter__(self):
+        r"""Iterates over the intervals. Will return a tuple of (start, end).
+        This iterator will not include other optional attributes.
+
+        .. Example ::
+
+            >>> import numpy as np
+            >>> from temporaldata import Interval
+
+            >>> intervals = Interval(
+            ...     start=np.array([0., 1., 2.]),
+            ...     end=np.array([1., 2., 3.]),
+            ...     some_other_attribute=np.array([0, 1, 2]),
+            ... )
+
+            >>> for start, end in intervals:
+            ...     print(start, end)
+            0.0 1.0
+            1.0 2.0
+            2.0 3.0
+        """
+        for s, e in zip(self.start, self.end):
+            yield (s, e)
```

Comment:
How about just returning `zip(self.start, self.end)`?
No strong feelings about doing it either way - returning the iterator, or yielding values. Just thinking if there's a reason to go one way or the other.

Reply:
Okay I don't think it matters at all. I'll merge this.